test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -82,12 +101,15 @@ TEST_F(XdsOverrideHostTest, SwapChildPolicy) {
   auto subchannel =
       FindSubchannel({kAddresses[0]},
                      ChannelArgs().Set(GRPC_ARG_INHIBIT_HEALTH_CHECKING, true));
-  ASSERT_TRUE(subchannel);
+  ASSERT_NE(subchannel, nullptr);
   ASSERT_TRUE(subchannel->ConnectionRequested());
+  subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
+  subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
+  ASSERT_NE(WaitForConnected(), nullptr);
```

Comment:
Please check the behavior of the picker instead of just asserting that it's non-null.